File: Pascal/step01/tictactoe.pas
program TicTacToe;

{ Tic-Tac-Toe Game - Data Structures and Variables }

type
  { Board cell can be empty, X, or O }
  TCellState = (Empty, X, O);
  
  { 3x3 game board }
  TBoard = array[1..3, 1..3] of TCellState;
  
  { Game state enumeration }
  TGameState = (Playing, XWins, OWins, Draw);

var
  { Main game board }
  Board: TBoard;
  
  { Current player (X starts first) }
  CurrentPlayer: TCellState;
  
  { Current game state }
  GameState: TGameState;
  
  { Move counter to track number of moves made }
  MoveCount: Integer;

{ Procedure to initialize the game board and variables }
procedure InitializeGame;
var
  i, j: Integer;
begin
  { Initialize all board cells to empty }
  for i := 1 to 3 do
    for j := 1 to 3 do
      Board[i, j] := Empty;
  
  { Set starting player to X }
  CurrentPlayer := X;
  
  { Set initial game state }
  GameState := Playing;
  
  { Reset move counter }
  MoveCount := 0;
end;

begin
  { Initialize the game when program starts }
  InitializeGame;
  
  { Program ends here for now - no gameplay logic yet }
end.

File: Ruby/step01/tic_tac_toe.rb
# The game board, represented as a 3x3 grid.
# Initially, each cell is numbered 1-9.
@board = [
  ["1", "2", "3"],
  ["4", "5", "6"],
  ["7", "8", "9"]
]

# The current player, either "X" or "O".
@current_player = "X"

# The state of the game.
# Can be :in_progress, :winner, or :draw.
@game_state = :in_progress
